From: "K. Y. Srinivasan" <kys@microsoft.com>
Date: Mon, 3 Feb 2014 12:42:45 -0800
Subject: Drivers: hv: vmbus: Support per-channel driver state
Patch-mainline: submitted (lkml 2014-02-03)

As we implement Virtual Receive Side Scaling on the networking side
(the VRSS patches are currently under review), it will be useful to have
per-channel state that vmbus drivers can manage. Add support for
managing per-channel state.

Signed-off-by: K. Y. Srinivasan <kys@microsoft.com>
Acked-by: <ohering@suse.de>
---
 include/linux/hyperv.h | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/include/linux/hyperv.h b/include/linux/hyperv.h
index 03443ff..35d84ba 100644
--- a/include/linux/hyperv.h
+++ b/include/linux/hyperv.h
@@ -1089,6 +1089,10 @@ struct vmbus_channel {
 	 * This will be NULL for the primary channel.
 	 */
 	struct vmbus_channel *primary_channel;
+	/*
+	 * Support per-channel state for use by vmbus drivers.
+	 */
+	void *per_channel_state;
 };
 
 static inline void set_channel_read_state(struct vmbus_channel *c, bool state)
@@ -1096,6 +1100,16 @@ static inline void set_channel_read_state(struct vmbus_channel *c, bool state)
 	c->batched_reading = state;
 }
 
+static inline void set_per_channel_state(struct vmbus_channel *c, void *s)
+{
+	c->per_channel_state = s;
+}
+
+static inline void *get_per_channel_state(struct vmbus_channel *c)
+{
+	return c->per_channel_state;
+}
+
 void vmbus_onmessage(void *context);
 
 int vmbus_request_offers(void);
